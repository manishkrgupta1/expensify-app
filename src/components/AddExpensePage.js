import React from 'react';
import ExpenseForm from './ExpenseForm';
import {connect} from  'react-redux';
import {startAddExpense} from '../actions/expenses';


 export const AddExpensePage =(props) =>(
    <div>
    <h1>Add Expense</h1>
    <ExpenseForm 
      onSubmit={(expense)=>{
        //props.dispatch(addExpense(expense));
        props.startAddExpense(expense);
        props.history.push('/');
        //used to redirect to dash board after form submit
      }}
    />
    </div>
  );

  const mapDispatchToProps =(dispatch)=>({
         startAddExpense :(expense)=> dispatch(startAddExpense(expense))
  });
  
  export default connect(undefined, mapDispatchToProps)(AddExpensePage);  