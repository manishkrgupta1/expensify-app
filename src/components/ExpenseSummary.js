import React from 'react';
import {connect} from 'react-redux';
import selectExpenses from '../selectors/expenses';
import expenseTotal from '../selectors/expenseTotal';
import numeral from 'numeral';

export const ExpenseSummary =({expenseCount , expenseTotal})=>{
    const expenseWord = expenseCount == 1 ?'expense':'expenses';
    const formattedTotalExpenses=numeral(expenseTotal /100).format('$0,0.00');
    return(
     <div>
      {
            <h1>{`viewing ${expenseCount} ${expenseWord} totaling ${formattedTotalExpenses}`}</h1>
      }
     </div>
        );

};

const mapStateToProps=(state) =>{
    const visibleExpenses =selectExpenses(state.expenses , state.filters);
    return {
       expenseCount :visibleExpenses.length ,
       expenseTotal : expenseTotal(visibleExpenses) 
    };
};

export default connect(mapStateToProps)(ExpenseSummary);