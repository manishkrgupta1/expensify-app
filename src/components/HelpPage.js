import React from 'react';

const HelpPage =() =>(
    <div>
      This is my helpPage component
    </div>
  );
  export default HelpPage;  