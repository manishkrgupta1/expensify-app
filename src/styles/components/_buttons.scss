.button {
  background: $blue;
  border: none;
  display: inline-block;
  padding: $s-size;
  font-weight: 300;
  color: whitesmoke;
  font-size: $m-size;
  text-decoration: none;
  line-height: 1;
}
.button--link {
  background: none;
}
.button--remove {
  background: #888;
}
