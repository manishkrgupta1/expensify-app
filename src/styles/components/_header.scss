.header {
  background: $dark-blue;

  padding: $m-size;
}
.header__items {
  display: flex;
  justify-content: space-between;
  align-items: center;
}
.header__title {
  color: white;
  text-decoration: none;
  h1 {
    margin: 0;
  }
}
