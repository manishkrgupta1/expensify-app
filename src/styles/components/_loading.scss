.loader {
  height: 100vh;
  width: 100vw;
  display: flex;
  align-items: center;
  justify-content: center;
}

.loader__image {
  height: 5rem;
  width: 5rem;
}
