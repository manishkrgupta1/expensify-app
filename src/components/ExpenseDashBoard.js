import React from 'react';
import ExpenseList from './ExpenseList';
import ExpenseListFilters from './expenseListFilters';
import ExpenseSummary from './ExpenseSummary';

const ExpenseDashBoard =() =>(
    <div>
      <ExpenseSummary />
      <ExpenseListFilters/>
      <ExpenseList/>
    </div>
  );
  
  export default ExpenseDashBoard;  