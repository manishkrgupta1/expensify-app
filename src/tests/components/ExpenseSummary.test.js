import React from 'react';
import {shallow} from 'enzyme';
import {ExpenseSummary} from '../../components/ExpenseSummary';


test ('should correctly render ExpensesSummary with one expense',()=>{
    const wrapper=shallow(<ExpenseSummary expenseCount={1} expenseTotal={222}/>)
    expect(wrapper).toMatchSnapshot();

});

test ('should correctly render expense Summary with multple expenses',()=>{
    const wrapper=shallow(<ExpenseSummary expenseCount={2} expenseTotal={224542}/>)
    expect(wrapper).toMatchSnapshot();    

});