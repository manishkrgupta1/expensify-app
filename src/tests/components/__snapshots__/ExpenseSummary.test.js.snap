// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render ExpensesSummary with one expense 1`] = `
<div>
  <h1>
    viewing 1 expense totaling $2.22
  </h1>
</div>
`;

exports[`should correctly render expense Summary with multple expenses 1`] = `
<div>
  <h1>
    viewing 2 expenses totaling $2,245.42
  </h1>
</div>
`;
